java/developer-guide/working-with-outlook-items/working-with-recurrences/_index.md

```diff
@@ -0,0 +1,527 @@
+
+1. Set start, end and due date.
+1. Create a MapiTask.
+1. Set task state to NotAssigned.
+1. Create the monthly recurrence object by setting the properties like PatternType, Period, WeekStartDay, EndType and OccurenceCount.
+1. Set MapiTask.Recurrence property to this monthly recurrence object.
+1. Save this message on disc.
+
+The following code snippet shows you how to create task with recurrence end type as EndAfterNOccurrence.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-.NET
+// The path to the File directory.
+String dataDir = RunExamples.getDataDir_Outlook();
+
+Date startDate = getDate(2015, 7, 16);
+Date DueDate = getDate(2015, 7, 16);
+Date endByDate = getDate(2015, 12, 31);
+
+MapiTask task = new MapiTask("This is test task", "Sample Body", startDate, dueDate);
+task.setState(MapiTaskState.NotAssigned);
+
+// Set the Monthly recurrence
+MapiCalendarMonthlyRecurrencePattern rec = new MapiCalendarMonthlyRecurrencePattern();
+rec.setDay(15);
+rec.setPeriod(1);
+rec.setPatternType(MapiCalendarRecurrencePatternType.Month);
+rec.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+rec.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=MONTHLY;BYMONTHDAY=15;INTERVAL=1"));
+rec.setWeekStartDay(DayOfWeek.Monday);
+
+if (rec.getOccurrenceCount() == 0)
+{
+    rec.setOccurrenceCount(1);
+}
+
+task.setRecurrence(rec);
+task.save(dataDir + "Monthly_out.msg", TaskSaveFormat.Msg);
+~~~
+
+
+
+Following function can be used to calculate the number of events between the two dates:
+
+
+
+~~~Java
+private static long getOccurrenceCount(Date start, Date endBy, String rrule)
+{
+    DateFormat formatter = new SimpleDateFormat("yyyyMMdd");
+    CalendarRecurrence pattern = new CalendarRecurrence("DTSTART:" + formatter.format(start) + "\r\nRRULE:" + rrule);
+    DateCollection dates = pattern.generateOccurrences(start, endBy);
+    return dates.size();
+}
+~~~
+#### **Set fix number of occurrences**
+The following code snippet shows you how to set fix number of occurrences.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the Monthly recurrence
+MapiCalendarMonthlyRecurrencePattern records = new MapiCalendarMonthlyRecurrencePattern();
+records.setDay(15);
+records.setPeriod(1);
+records.setPatternType(MapiCalendarRecurrencePatternType.Month);
+records.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+records.setOccurrenceCount(5);
+records.setWeekStartDay(DayOfWeek.Monday);
+~~~
+### **Monthly Recurrences: EndAfterDate Recurrance Type**
+"End By" option in the Mapi Task is achieved by setting the OccurrenceCount property calculated by the GetOccurrenceCount() function. This function takes start date , end date and RRULE string. The following code snippet shows you how to creates a recurrence on 15th of each month between start and end by date.
```

Comment:
`*OccurrenceCount* property calculated by the **getOccurrenceCount()** function.`